### ISSUE TITLE
Fix Layout/RedundantLineBreak issue with anonymous splat forwarding

### ISSUE THREAD

**Issue by bkuhlmann** [rubocop/rubocop#11411]
## Expected behavior

Hello. :wave: When using the following configuration:

``` yaml
Layout/RedundantLineBreak:
  Enabled: true
  InspectBlocks: true
```

I'd like to Rubocop to not flag single (`*`) or double (`**`) splats as violations.

## Actual behavior

I'm seeing false positives with `Layout/RedundantLineBreak` and `Layout/MultilineOperationIndentation` when forwarding arguments.

## Steps to reproduce the problem

Use the following code:

``` ruby
class One
  def initialize(handler = Object.new, *)
    super *
    @handler = handler
  end
end

class Two
  def initialize(handler: Object.new, **)
    super **
    @handler = handler
  end
end
```

When you use RuboCop to analyze the above, you'll end up with the following errors:

```
Offenses:

/Users/bkuhlmann/Engineering/Misc/snippet:17:5: C: [Correctable] Layout/RedundantLineBreak: Redundant line break detected.
    super * ...
    ^^^^^^^
/Users/bkuhlmann/Engineering/Misc/snippet:18:5: C: [Correctable] Layout/MultilineOperationIndentation: Use 2 (not 0) spaces for indenting an expression spanning multiple lines.
    @handler = handler
    ^^^^^^^^^^^^^^^^^^
/Users/bkuhlmann/Engineering/Misc/snippet:24:5: C: [Correctable] Layout/RedundantLineBreak: Redundant line break detected.
    super ** ...
    ^^^^^^^^
/Users/bkuhlmann/Engineering/Misc/snippet:24:11: C: [Correctable] Layout/SpaceAroundOperators: Space around operator ** detected. (https://rubystyle.guide#spaces-operators)
    super **
          ^^
/Users/bkuhlmann/Engineering/Misc/snippet:25:5: C: [Correctable] Layout/MultilineOperationIndentation: Use 2 (not 0) spaces for indenting an expression spanning multiple lines.
    @handler = handler
    ^^^^^^^^^^^^^^^^^^

1 file inspected, 5 offenses detected, 5 offenses autocorrectable
```

By the way, if you were to let RuboCop autocorrect the above code, you'd get the following source code which is syntactically incorrect:

``` ruby
class One
  def initialize(handler = Object.new, *)
    super * @handler = handler
  end
end

class Two
  def initialize(handler: Object.new, **)
    super(**@handler = handler)
  end
end
```

## RuboCop version

```
1.42.0 (using Parser 3.1.3.0, rubocop-ast 1.24.1, running on ruby 3.2.0) [arm64-darwin22.2.0]
  - rubocop-performance 1.15.2
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.16.0
  - rubocop-thread_safety 0.4.4
```

**Comment by koic** [comment]
This does not appear to be a false positive. Parentheses are required if anonymous args and anonymous rest args have the following lines:

```diff
class One
  def initialize(handler = Object.new, *)
-   super *
+   super(*)
    @handler = handler
  end
end

class Two
  def initialize(handler: Object.new, **)
-   super **
+   super(**)
    @handler = handler
  end
end
```

So the following are equivalent:

```ruby
super *
do_something

super * do_something
```

It's a Ruby syntax issue similar to the issue below:
https://bugs.ruby-lang.org/issues/18396

**Comment by bkuhlmann** [comment]
Ah, yes, I was too hasty in logging this issue. I apologize. I opened up a properly defined issue for this here: #11418.

By the way, thanks for fixing #11303. I can't upgrade to try this out yet until Reek updates to the new parser but looking forwarding to making use of this. :bow:
